src/database-layer/db.ts

```diff
@@ -529,6 +535,20 @@ if (maybePg != null) {
 				}
 			}
 
+			public override async getTxLevelLock(key1: number, key2: number) {
```

Comment:
I'd prefer to see the first key be pre-registered, eg how open-balena-api uses `ADVISORY_LOCK_NAMESPACES`, you'd pre-register your namespaces and then just provide the namespace key for the first argument

Reply:
Should that be done by a `registerLockNamespace` method exported by `database-layer/db` or would you imagine it be better if it's added as an extra property in `init`'s `config` object (in `server-glue/module`).
At first a `registerLockNamespace()` that each feature can register it's own namespaces sounded like more appropriate for the feature based separation that we have started following, but on second thought, we will still need coordination in order for two features to not end up with a namespace id clash.